app/services/solidus_bolt/transaction/void.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  module Transaction
+    class Void < SolidusBolt::Base
+      def initialize(transaction_reference:, credit_card_transaction_id:)
+        @transaction_reference = transaction_reference
+        @credit_card_transaction_id = credit_card_transaction_id
+        super
+      end
+
+      def call
+        capture
+      end
+
+      private
+
+      def capture
+        options = build_options
+        HTTParty.post(
+          "#{api_base_url}/#{api_version}/merchant/transactions/void",
+          options
+        )
+      end
+
+      def build_options
+        options = {
+          query: {
+            transaction_reference: @transaction_reference,
+            credit_card_transaction_id: @credit_card_transaction_id,
+          },
+          headers: {
+            'X-Nonce' => generate_nonce
+          }
+        }
+
+        authenticate(options)
+        options
+      end
+
+      def generate_nonce
+        SecureRandom.hex
+      end
+
+      def authenticate(options)
+        options[:headers].merge!(authentication_header)
+      end
```

Comment:
These 2 methods seem repeated in all the transaction objects. Maybe you can move them into SolidusBolt::Base, or in an included module.

Reply:
moved to base